### PR TITLE
[Runtime] use Symfony Runtime

### PR DESCRIPTION
**Pull request by dpfaffenbauer**

### CODE REVIEW

File: public/index.php [hunk 1]
```diff
@@ -1,39 +1,25 @@
...
-/** @var \Pimcore\Kernel $kernel */
-$kernel = \Pimcore\Bootstrap::kernel();
+    // set current request as property on tool as there's no
+    // request stack available yet
+    Tool::setCurrentRequest($request);
 
-// reset current request - will be read from request stack from now on
-Tool::setCurrentRequest(null);
+    /** @var \Pimcore\Kernel $kernel */
+    $kernel = \Pimcore\Bootstrap::kernel();
```
Comment (jdreesen) [inline public/index.php]:
```suggestion
    $kernel = Bootstrap::kernel();
```

File: bin/console [hunk 2]
```diff
@@ -1,49 +1,26 @@
...
-if (file_exists($a = getcwd() . '/vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../../../../vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../autoload.php')) {
-    include $a;
+use App\Kernel;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
```
Comment (jdreesen) [inline bin/console]: Is it right, that this is no `\Pimcore\Console\Application` anymore?

File: bin/console [hunk 3]
```diff
@@ -1,49 +1,26 @@
...
-
-if (file_exists($a = getcwd() . '/vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../../../../vendor/autoload.php')) {
-    include $a;
-} elseif (file_exists($a = __DIR__ . '/../autoload.php')) {
-    include $a;
+use App\Kernel;
```
Comment (jdreesen) [inline bin/console]: This seems to be unused.

File: public/index.php [hunk 4]
```diff
@@ -1,39 +1,25 @@
...
- *
- * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
- * @license    http://www.pimcore.org/license     GPLv3 and PEL
- */
-
+use Pimcore\Bootstrap;
 use Pimcore\Tool;
 use Symfony\Component\HttpFoundation\Request;
 
-include __DIR__ . "/../vendor/autoload.php";
```
Comment (jdreesen) [inline public/index.php:19]: The `Bootstrap::bootstrap()` also loads the `vendor/autoload.php`. Does it interfere somehow, or is it safe?

File: public/index.php [hunk 5]
```diff
@@ -1,39 +1,25 @@
...
-
-\Pimcore\Bootstrap::setProjectRoot();
-\Pimcore\Bootstrap::bootstrap();
+require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
```
Comment (jdreesen) [inline public/index.php:19]: The `Bootstrap::bootstrap()` also loads the `vendor/autoload.php`. Does it interfere somehow, or is it safe?

**Comment by brusch** [comment]
See also https://github.com/pimcore/pimcore/issues/10344 (probably resolves it), needs to be considered here as well: 

- https://github.com/pimcore/demo
- https://github.com/pimcore/demo-enterprise

File: public/index.php (hunk 5 above)
Comment (dvesh3) [reply public/index.php:19]: please see https://github.com/pimcore/pimcore/issues/14485

File: bin/console (hunk 2 above)
Comment (dvesh3) [reply bin/console]: we should still use `\Pimcore\Console\Application` here as it does couple of things:
- set locale for `umlauts` & `escapeshellarg()`
- respect maintenance mode
- add `prefix` option to filter for version classes on DoctrineCommand

**Comment by dvesh3** [comment]
We can continue working on this PR to leverage Symfony Runtitme. Only need some refactorings on the core regarding bootstrapping, which will be covered in https://github.com/pimcore/pimcore/issues/14485

File: bin/console (hunk 3 above)
Comment (dvesh3) [reply bin/console]:
```suggestion
```

File: public/index.php [hunk 6]
```diff
@@ -1,39 +1,24 @@
...
- * - GNU General Public License version 3 (GPLv3)
- * - Pimcore Enterprise License (PEL)
- * Full copyright and license information is available in
- * LICENSE.md which is distributed with this source code.
- *
- * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
- * @license    http://www.pimcore.org/license     GPLv3 and PEL
- */
-
+use Pimcore\Bootstrap;
```
Comment (dvesh3) [inline public/index.php:15]: please keep license comments

File: bin/console [hunk 7]
```diff
@@ -14,36 +13,25 @@
...
+    require_once $a;
+} elseif (file_exists($a = __DIR__ . '/../autoload_runtime.php')) {
+    require_once $a;
 } else {
-    fwrite(STDERR, 'Cannot locate autoloader; please run "composer install"' . PHP_EOL);
-    exit(1);
+    throw new LogicException('Symfony Runtime is missing. Try running "composer require symfony/runtime".');
 }
 
 \Pimcore\Bootstrap::setProjectRoot();
```
Comment (jdreesen) [inline bin/console:30]: What about `use Pimcore\Bootstrap;` like in the `public/index.php`?

**Review by dvesh3 — APPROVED** [review]

File: bin/console (hunk 7 above)
Comment (dvesh3) [reply bin/console:30]: @dpfaffenbauer one last change then we are ready to merge 💪

**Pull request merged**